Feature leo

leo491427 opened the pull request:
Dear Bear,

I found that your original code had a minor issue on pagination page count when you search by a key word.  I've adjusted the code a little bit to eliminate this minor issue. Please check it.

The idea of my adjustment is fetching the number of eligible documents according to the search condition before pagination, rather than fetching the number of all the documents.

Your code is organized and written in a perfect way. Some parts are beyond my ability to undertand thoroughly, so my adjustment may not be in the best way. If you have any thoughts, please feel free to tell me.  Thanks.

leo491427 committed:
fixed pagination count

leo491427 committed:
fixed pagination count v1.1

LazeBear commented on src/services/service.js:
```diff
@@ -25,6 +25,20 @@ class Service {
+    let query, count;
+
+    if (search) {
+      query = await this.Model.searchByKeyword(search);
```
so you want to get the count right, then why not just find the count, instead of find all the documents then get the count. Have a look of the docs for mongodb and mongoose

LazeBear reviewed (commented):
please make another pr after you updated your code. contact me if needed

leo491427 committed:
fixed pagination count v1.2 + adjusted logger setting

leo491427 committed:
fixed pagination count v1.3 + adjusted logger setting

leo491427 commented:
Hi Bear,

I pulled a new request. Please check the latest commit "fixed pagination count v1.3 + adjusted logger setting".

I updated according to your comments. I've no questions on your 1st and 2nd comment now. As for your 3rd comment, I changed the way of counting when there isn't any specific search. But I cannot change the way of counting when a specific search is needed, as the return of the function "searchByKeyword" has already been a documents array. To be honest, I wrote this function "searchByKeyword" according to your original one "searchQuery", and I don't know how to make it return a query instead of an array.

If you could give any suggestinos, that would be really helpful. Thanks.
